### PR TITLE
Improve documentation about generating schema files.

### PR DESCRIPTION
**Pull request by ezekiiel**
Based on feedback from Whisky.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -55,25 +55,64 @@ From the repository root.
...
 
-To generate schema files, run:
+### Generating schema files
 
+We generate JSON schema files for all of our contracts' query and
+execute messages. We use those schema files to generate Typescript
+code for interacting with those contracts via
+[ts-codegen](https://github.com/CosmWasm/ts-codegen). This generated
+Typescript code is then [used in the
+UI](https://github.com/DA0-DA0/dao-dao-ui/tree/40f3cbfe676a98bf7b9db7b646e74e5b2dae4502/packages/state/clients).
```
Comment (elsehow) [inline CONTRIBUTING.md:67]: "Used in the UI to...." Important to tell people why they need to do this.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -55,25 +55,64 @@ From the repository root.
...
+
+To make sure these files are generated correctly, make sure to add any
+new data types used in contract messages to `examples/schema.rs`.
+
+If you are adding a new query ts-codegen expects:
```
Comment (elsehow) [inline CONTRIBUTING.md:72]: > If you are adding a new query, `ts-codegen` expects:

**Review by elsehow — APPROVED** [review]
Looks great! A few minor comments. Otherwise, ready to go!

File: CONTRIBUTING.md (hunk 1 above)
Comment (ezekiiel) [reply CONTRIBUTING.md]: Great feedback. Added a blurb about frontend folks not needing to manually write query and execute messages.

**Review by blue-note — APPROVED** [review]
Awesome! This is very helpful

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -55,25 +55,67 @@ From the repository root.
 Before making a PR, you'll need to do two things to get CI passing:
 
 1. Generate schema files for the contracts.
```
Comment (blue-note) [inline CONTRIBUTING.md:57]: should we also make a note about running `cargo schema`?

Comment (ezekiiel) [reply CONTRIBUTING.md:57]: I think this is covered by the earlier section that mentions running `scripts/schema.sh` which does that under the hood.

Comment (ezekiiel) [reply CONTRIBUTING.md:57]: oh shit wait you're right the order is wrong..

Comment (ezekiiel) [reply CONTRIBUTING.md:57]: or wait no this is correct. This section is giving an overview, but the end result is just prompting the reader to run the `schema.sh` script.

**Pull request merged**